typescript/utils/src/objects.ts

```diff
@@ -175,3 +179,35 @@ export function normalizeConfig(obj: WithAddress<any>): any {
 
   return obj;
 }
+
+// write a function that will go through an object and sort any arrays it finds
+export function sortValidatorsAndModulesInConfig(config: any): any {
+  // Check if the current object is an array
+  if (Array.isArray(config)) {
+    return config.map(sortValidatorsAndModulesInConfig);
+  }
+  // Check if it's an object and not null
+  else if (typeof config === 'object' && config !== null) {
+    const sortedConfig: any = {};
+    for (const key in config) {
+      if (key === 'validators' && Array.isArray(config[key])) {
+        // Sort the validators array in lexicographical order (since they're already lowercase)
+        sortedConfig[key] = config[key].sort();
+      }
+      // if key is modules, sort the objects in the array by their 'type' property
+      else if (key === 'modules' && Array.isArray(config[key])) {
```

Comment:
what about hooks in the hookconfig? https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/74a592e5867a375f1705654c1112d7c9ac784ddf/typescript/sdk/src/hook/EvmHookReader.ts#L193